torch/csrc/jit/graph_node_list.h

```diff
@@ -114,10 +114,10 @@ struct generic_graph_node_list {
     return generic_graph_node_list_iterator<const T>(head->next_in_graph[d], d);
   }
   generic_graph_node_list_iterator<T> end() {
-    return generic_graph_node_list_iterator<T>(head, d);
+    return generic_graph_node_list_iterator<T>(head->next_in_graph[!d], d);
```

Comment:
This excludes the node prior to `head` from the iteration. It's a bit non-standard for containers to do this, so I'd rather have us slice the `graph_node_list` inside `Graph` than resort to hacks like this.

Reply:
I am not sure exactly what you are asking for here. Can you clarify? 

In this change the node prior to head is always tail. Both head and tail are suppose to be excluded from the list.

Before and after this pass, the node_list always represents the full list and not a slice. It has the property that it cannot be invalidated no matter what inserts/deletes would happen in the list because it doesn't hold a reference to a node in the list.